src/main/java/net/countercraft/movecraft/combat/movecraftcombat/directors/CannonDirectorManager.java

```diff
@@ -154,34 +155,27 @@ private void processDirectors() {
                 if (p == null || p.getInventory().getItemInMainHand().getType() != Config.DirectorTool) {
                     continue;
                 }
-                Vector tv = tnt.getVelocity();
-                double speed = tv.length(); // store the speed to add it back in later, since all the values we will be using are "normalized", IE: have a speed of 1
-                tv = tv.normalize(); // you normalize it for comparison with the new direction to see if we are trying to steer too far
+                Vector tntVector = tnt.getVelocity();
+                // Store the speed to add it back in later, since all the values we will be using are "normalized", IE: have a speed of 1
+                // We're only interested in the horizontal speed for now since that's all directors *should* affect.
+                double horizontalSpeed = new Vector(tntVector.getX(),0,tntVector.getY()).length();
+                tntVector = tntVector.normalize(); // you normalize it for comparison with the new direction to see if we are trying to steer too far
                 Block targetBlock = p.getTargetBlock(Config.Transparent, Config.CannonDirectorRange);
                 Vector targetVector;
-                if (targetBlock == null) { // the player is looking at nothing, shoot in that general direction
+                // This is horrible but I'm not sure how else to do it, targetBlock == null doesn't work because getTargetBlock never returns null.
+                if (targetBlock.getType().equals(Material.AIR)) { // the player is looking at nothing, shoot in that general direction
                     targetVector = p.getLocation().getDirection();
                 } else { // shoot directly at the block the player is looking at (IE: with convergence)
                     targetVector = targetBlock.getLocation().toVector().subtract(tnt.getLocation().toVector());
-                    targetVector = targetVector.normalize();
```

Comment:
This vector.normalize call is unnecessary as the vector is normalized a few lines down